README.md

```diff
@@ -1,135 +1,268 @@
+which and is a well-known and practically exploited vulnerability.
+
+Identifying parser differentials and exploits is often a time-consuming
+task involving multiple iterations of hypothesizing, developing
+special-purpose tools, running against a file corpora and analysing the
+results. Reducing the manual aspects of this cycle can serve to
+significantly accelarate this loop and enable investigations on much
+larger file corpora and with more parsers and tools than manually
+possible. Further, such scaling also enables the discovery of
+statistical correlations between the outputs of the tools which can help
+to surface the root causes of many differentials and exploits.
+
+The Feature Analysis Workbench (FAW) provides a workbench for
+file-format analysis that enables users to easily create and integrate
+new as well as existing tools and parsers for formats and compare and
+contrast their behavior against large file corpora scaling up to
+millions of files. The FAW has been designed as a flexible tool to
+examine and understand data formats and allow different types of users
+to work with these formats from different perspectives. For example:
+
+- Specification writers: the FAW makes it easy to compare various
+  parsers that implement the spec as well as examine the kind of
+  documents that are available in the wild and how they interpret the
+  specification.
+- Parser Developers: in addition to the parser comparisons mentioned
+  above, the FAW also makes it easy to make changes to a parser and test
+  parser behavior against large corpus.
+- Vulnerability Researchers: the FAW allows easy incorporation of new
+  ananlysis to dig more in to the programs and the specification under
+  test.
+
+# Screenshot
 
 Main overview of decisions
 ![Main overview](./docs/readme/screenshot-main.png)
 
-Decision DSL excerpt
-![DSL excerpt](./docs/readme/screenshot-dsl.png)
+# Architecture of the FAW
+
+Exploring data formats via input corpora at scale requires an
+infrastructure to run parsers and associated tools against input files
+in the corpus and capture their behavior. The FAW is divided in to a
+backend that handles the details of running parsers against the input
+corpus, and a frontend that enables users to investigate the behavior of
+parsers and tools against files in the corpus, at different scales and
+granularities.
+
+Parsers and parser-like tools are provided as binary executables and can
+be written in any language, and using any technology. Further FAW
+supports processing the file corpus either on a single machine or in
+parallel on a cluster of machines. The latter, termed a "teaming" setup,
+can be performed on a private network without requiring internet access.
+
+The FAW allows the user to specify analysis sets to control the scale
+and speed of analyses. It also provides multiple tools are provided to
+explore the interactions between inputs and parsing programs in the
+analysis set.
+
+TODO: Note on features as booleans.
```

Comment:
This might best be its own little section, with a link from the second paragraph to more details about the data representation

Reply:
A new section has been added (in commit [50ba392](https://github.com/GaloisInc/FAW/pull/94/commits/50ba3924ba4af7a97a1003fd15c76a0d80b4c03b)) with this.